events/recorder.go

```diff
@@ -0,0 +1,78 @@
+// Recorder
+//
+// The Recorder interface provides an interface for
+// workloads and operations to collect data about their internal
+// state, without requiring workloads to be concerned with data
+// retention, storage, or compression. The implementations of Recorder
+// provide different strategies for data collection and persistence
+// so that tests can easily change data collection strategies without
+// modifying the test.
+package events
+
+import "time"
+
+// Recorder describes an interface that tests can use to track metrics
+// and events during performance testing or normal
+// operation. Implementations of recorder wrap an FTDC collector and
+// will write data out to the collector for reporting purposes. The
+// types produced by the collector use the Performance or
+// PerformanceHDR types in this package.
+//
+// Choose the implementation of Recorder that will capture all
+// required data used by your test with sufficient resolution for use
+// later. Additional consider the data volume produced by the
+// recorder.
+type Recorder interface {
+	// The Inc<> operations add values to the specified counters
```

Comment:
This is probably not godoc-friendly, which may be important to you.

Reply:
it isn't.